x-pack/plugins/viz_editor/public/common/components/operation_editor/operation_definitions.tsx

```diff
@@ -0,0 +1,431 @@
+  const opDefinition = getOperationDefinition(column.operation);
+  const options = opDefinition
+    .applicableFields(visModel.datasource.fields, props)
+    .map((f: DatasourceField) => ({
+      value: f.name,
+      text: f.name,
+    }));
+
+  return (
+    <EuiSelect
+      options={options}
+      value={argument && argument.field}
+      onChange={e =>
+        onColumnChange({
+          ...operation,
+          alias: e.target.value,
+          argument: {
+            ...argument,
+            field: e.target.value,
+          },
+        } as SelectOperation)
+      }
+      aria-label="Field"
+    />
+  );
+}
+
+function windowOperationEditor(props: OperationEditorProps) {
+  const { column, visModel, onColumnChange } = props;
+  const operation = column as WindowOperation;
+  const { argument } = operation;
+  const opDefinition = getOperationDefinition(column.operation);
+  const options = opDefinition
+    .applicableFields(visModel.datasource.fields, props)
+    .map((f: DatasourceField) => ({
+      value: f.name,
+      text: f.name,
+    }));
+  const calculations = [
+    { value: 'max', text: 'Max' },
+    { value: 'min', text: 'Min' },
+    { value: 'sum', text: 'Sum' },
+    { value: 'stdDev', text: 'Standard Deviation' },
+    { value: 'unweightedAvg', text: 'Unweighted Average' },
+    { value: 'linearWeightedAvg', text: 'Linear Weighted Average' },
+    { value: 'ewma', text: 'Ewma' },
+    { value: 'holt', text: 'Holt' },
+    { value: 'holtWinters', text: 'Holt Winters' },
+  ];
+
+  return (
+    <EuiForm>
+      <EuiFormRow label="Field">
+        <EuiSelect
+          options={options}
+          value={argument && argument.field}
+          onChange={e =>
+            onColumnChange({
+              ...operation,
+              argument: {
+                ...argument,
+                field: e.target.value,
+              },
+            })
+          }
+          aria-label="Field"
+        />
+      </EuiFormRow>
+      <EuiFormRow label="Calculation">
+        <EuiSelect
+          options={calculations}
+          value={argument && argument.windowFunction}
+          onChange={e =>
+            onColumnChange({
+              ...operation,
+              argument: {
+                ...argument,
+                windowFunction: e.target.value as WindowFunction,
+              },
+            })
+          }
+          aria-label="Window Function"
+        />
+      </EuiFormRow>
+      <EuiFormRow label="Window size">
+        <EuiFieldNumber
+          value={argument && argument.windowSize}
+          onChange={e =>
+            onColumnChange({
+              ...operation,
+              argument: {
+                ...argument,
+                windowSize: parseInt(e.target.value, 10),
+              },
+            })
+          }
+          aria-label="Window size"
+        />
+      </EuiFormRow>
+    </EuiForm>
+  );
+}
+
+export const operations: OperationDefinition[] = [
+  {
+    name: 'Values',
```

Comment:
Please use the `operationToName('column')` function to get consistent and i18n names

Reply:
Makes sense, thanks.